diff --git a/node.gypi b/node.gypi
index f6787e5ad8..021d2dc8cb 100644
--- a/node.gypi
+++ b/node.gypi
@@ -84,6 +84,15 @@
         }],
       ],
     }],
+    [ 'node_shared=="true" and node_use_bundled_v8=="false"', {
+      'link_settings': {
+        'libraries': [
+          '-lv8',
+          '-lv8_libbase',
+          '-lv8_libplatform',
+        ],
+      },
+    }],
     [ 'node_use_v8_platform=="true"', {
       'defines': [
         'NODE_USE_V8_PLATFORM=1',
