diff --git a/Makefile.in b/Makefile.in
index dd0fc57..f26b281 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -97,15 +97,11 @@ all: ccache$(EXEEXT)
 ccache$(EXEEXT): $(ccache_objs) $(extra_libs)
 	$(CC) $(all_cflags) -o $@ $(ccache_objs) $(LDFLAGS) $(extra_libs) $(LIBS)
 
-ccache.1: doc/ccache.1
-	cp $< $@
-
 .PHONY: install
-install: ccache$(EXEEXT) ccache.1
+install: ccache$(EXEEXT)
 	$(installcmd) -d $(DESTDIR)$(bindir)
 	$(installcmd) -m 755 ccache$(EXEEXT) $(DESTDIR)$(bindir)
 	$(installcmd) -d $(DESTDIR)$(mandir)/man1
-	-$(installcmd) -m 644 ccache.1 $(DESTDIR)$(mandir)/man1/
 
 .PHONY: clean
 clean:
